packages/azure-themes/src/azure/styles/DetailsList.styles.ts

```diff
@@ -49,6 +50,25 @@ export const DetailsListStyles = (props: IDetailsListStyleProps): Partial<IDetai
   return {
     root: {
       borderTop: StyleConstants.borderNone,
+      selectors: {
+        ':has(.ms-Check.is-checked) .ms-DetailsHeader:not(.is-allSelected) .ms-DetailsHeader-cellIsCheck': {
```

Comment:
This selector style on the DetailsList root implements the indeterminate state for the header checkbox. It will be provided as the default behavior, not as an opt-in via a prop per our in person discussion. This matches the behavior of the non-fluent details list in the Azure portal.


Here's how it looks:

|Theme||
|---|---|
|Light|<img width="322" alt="Screenshot 2024-02-08 at 4 18 00 PM" src="https://github.com/microsoft/fluentui/assets/79171711/aaa6934a-05d6-4ff5-a0e0-603d827211b8">|
|Dark|<img width="328" alt="Screenshot 2024-02-08 at 4 17 21 PM" src="https://github.com/microsoft/fluentui/assets/79171711/a0420f32-2379-4de8-92b6-13a3ee30e525">|
|High Contrast Light | <img width="321" alt="Screenshot 2024-02-08 at 4 17 27 PM" src="https://github.com/microsoft/fluentui/assets/79171711/636705b7-b62d-4e7c-8226-944a031eb341"> |
|High Contrast Dark | <img width="310" alt="Screenshot 2024-02-08 at 4 17 34 PM" src="https://github.com/microsoft/fluentui/assets/79171711/6d1fcda5-fe20-4aa3-a533-c7733ed0630e"> |

Reply:
Added to PR description